### DISCUSSION TITLE
Problem with JWT\Signer\Key

### DISCUSSION THREAD

**Discussion by TAURO-AG** [lcobucci/jwt#990] [Q&A] [answered]
The error: PHP Fatal error: Uncaught Error: Cannot instantiate interface Lcobucci\\JWT\\Signer\\Key

My program:
<pre>
require __DIR__ . '/vendor/autoload.php';
use Lcobucci\JWT\Signer\Rsa\Sha256;
use Lcobucci\JWT\Signer\Key;
use Lcobucci\JWT\Builder;

$host = '*****';
$private_key =  '*****';
$uid = '*****';

$time       = time();
$signer     = new Sha256();
$privateKey = new Key(base64_decode($private_key, true));
$token      = (new Builder())->identifiedBy(bin2hex(random_bytes(12)))
							 ->expiresAt($time + 1 * 86400)
							 ->getToken($signer, $privateKey);</pre>

**Comment by lcobucci** [comment, marked as answer]
@TAURO-AG the error is correct, from v4+ that code is invalid.

Please follow the [documentation](https://lcobucci-jwt.readthedocs.io/en/stable/) to make sure you have the expected code - pay close attention to the [upgrading guide](https://lcobucci-jwt.readthedocs.io/en/stable/upgrading/#v3x-to-v4x) if you're migrating from v3.x
